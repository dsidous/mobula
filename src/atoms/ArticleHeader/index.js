import React from 'react';
import { string, object } from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import TextFormatIcon from '@material-ui/icons/TextFormat';
import AppsIcon from '@material-ui/icons/Apps';
import StarBorderIcon from '@material-ui/icons/StarBorder';

const styles = () => ({
  icon: {
    fontSize: 12,
    marginRight: 6,
  },
});

const ArticleHeader = ({ type, mode, classes }) => (
  <>
    {type === 'text' && <TextFormatIcon className={classes.icon} />}
    {type === 'product' && <AppsIcon className={classes.icon} />}
    {type === 'favourite' && <StarBorderIcon className={classes.icon} />}
    <Typography>{`${mode} ${type}`}</Typography>
  </>
);

ArticleHeader.propTypes = {
  mode: string,
  type: string.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  classes: object.isRequired,
};

ArticleHeader.defaultProps = {
  mode: '',
};

export default withStyles(styles)(ArticleHeader);
