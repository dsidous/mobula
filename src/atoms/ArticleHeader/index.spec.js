import React from 'react';
import { shallow } from 'enzyme';

import ArticleHeader from '.';

describe('Atoms/ArticleHeader', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<ArticleHeader mode="Edit" type="Text" />);

    expect(wrapper).toMatchSnapshot();
  });
});
