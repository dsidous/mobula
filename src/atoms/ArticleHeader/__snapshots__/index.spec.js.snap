// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/ArticleHeader should render as expected 1`] = `ShallowWrapper {}`;
